FE/angular/src/app/components/movies/movies.component.ts

```diff
@@ -1,34 +1,29 @@
-import { Component, OnDestroy, OnInit } from '@angular/core';
-import { tap } from 'rxjs';
-import { DataService, MovieComplete } from '../../services/data.service';
+import { Component, OnInit } from '@angular/core';
+import { Observable, tap } from 'rxjs';
+import { DataService, MovieComplete, MovieData } from '../../services/data.service';
 
 @Component({
   selector: 'app-movies',
   templateUrl: './movies.component.html'
 })
-export class MoviesComponent implements OnDestroy, OnInit {
+export class MoviesComponent implements OnInit {
   public currDecade: number | undefined;
-  public decades: number[] = [];
+  // public decades: number[] = [];
   public filteredMovies: MovieComplete[] = [];
   public movies: MovieComplete[] = [];
-  private moviesSubscription: any;
+  public moviesSubscription: Observable<MovieData> = {} as Observable<MovieData>;
 
   constructor(private dataService: DataService) {}
 
   public ngOnInit(): void {
     this.moviesSubscription = this.dataService.getMovies().pipe(
       tap((data) => {
-        this.decades = data.Decades;
         this.movies = data.Search;
         this.displayMovies();
       })
     );
   }
 
-  public ngOnDestroy(): void {
-    this.moviesSubscription.unsubscribe();
-  }
-
   public displayMovies(decade?: number): void {
```

Comment:
Please be prepared to discuss:

* Why this approach is not "good" code ?
* What would be a better RxJS approach ?